File: SDK/java/src/main/java/com/reloop/verifio/VerifyService.java
package com.reloop.verifio;

import com.reloop.verifio.types.VerificationResult;
import com.reloop.verifio.types.VerifyOptions;

import java.util.HashMap;
import java.util.Map;

public class VerifyService {
    private final VerifioClient client;

    public VerifyService(VerifioClient client) {
        this.client = client;
    }

    public VerificationResult verify(String email, VerifyOptions options) {
        Map<String, Object> body = new HashMap<>();
        body.put("email", email);
        if (options != null) {
            body.put("options", options);
        }

        return this.client.request("POST", "/verify", body, VerificationResult.class);
    }
}

File: SDK/php/src/VerifyService.php
<?php

namespace Verifio;

use Verifio\Types\VerifyOptions;

class VerifyService
{
    private VerifioClient $client;

    public function __construct(VerifioClient $client)
    {
        $this->client = $client;
    }

    /**
     * @param string $email
     * @param VerifyOptions|null $options
     * @return array
     */
    public function verify(string $email, ?VerifyOptions $options = null): array
    {
        $body = ['email' => $email];
        if ($options !== null) {
            $body['options'] = $options->toArray();
        }

        return $this->client->request('POST', '/verify', $body);
    }
}
